src/components/PickupBox/PickupBox.jsx

```diff
@@ -34,27 +35,58 @@ const PickupBox = ({
   rejectionReason,
   fetchBoxes,
   pickup,
+  toast,
   imageStatus,
 }) => {
   // A state for determining whether or not the rejectBoxPopup is open
   // This state is set true when the reject button is clicked
   const [rejectBoxPopupIsOpen, setRejectBoxPopupIsOpen] = useState(false);
-
   // A function that updates the approved boolean in the backend and refreshes all boxes that are under review
   // This method is called when the approve box icon is clicked
+  const showToast = CustomToast({
+    icon: 'success',
+    title: `Box #${boxID} Approved`,
+    message: '',
+    toastPosition: 'bottom-right',
+  });
+
+  const errorToast = CustomToast({
+    icon: 'error',
+    title: `Failed to Approve Box #${boxID}`,
+    message: '',
+    toastPosition: 'bottom-right',
+  });
+
   const approvePickupBox = async id => {
-    await FYABackend.put('/boxHistory/approveBox', {
-      transactionID: id,
-    });
-    const requests = [
-      fetchBoxes('under review', true),
-      sendEmail(
-        boxHolderName,
-        boxHolderEmail,
-        renderEmail(<ApprovedBoxEmail boxHolderName={boxHolderName} />),
-      ),
-    ];
-    await Promise.all(requests);
+    try {
+      const boxId = id;
+      CustomToast(toast, {
+        icon: 'success',
+        title: `Box # Approved`,
+        message: '',
+        toastPosition: 'bottom-right',
+      });
+      FYABackend.put('/boxHistory/approveBox', {
+        boxID: boxId,
+      }).then(async () => {
+        await fetchBoxes('under review', true);
+        await FYABackend.put('/boxHistory/approveBox', {
+          transactionID: id,
+        });
+        const requests = [
+          fetchBoxes('under review', true),
+          sendEmail(
+            boxHolderName,
+            boxHolderEmail,
+            renderEmail(<ApprovedBoxEmail boxHolderName={boxHolderName} />),
+          ),
+        ];
+        await Promise.all(requests);
+        showToast();
+      });
```

Comment:
Not sure what happened here, but this function should look more like [the function](https://github.com/ctc-uci/find-your-anchor-frontend/blob/dev/src/components/PickupBox/PickupBox.jsx) in dev with an added try/catch. We shouldn't need the first FYABackend.put(...).then part. It should look something like this:

```
try {
   await FYABackend.put('/boxHistory/approveBox', {
      transactionID: id,
    });
    const requests = [
      fetchBoxes('under review', true),
      sendEmail(
        boxHolderName,
        boxHolderEmail,
        renderEmail(<ApprovedBoxEmail boxHolderName={boxHolderName} />),
      ),
    ];
    await Promise.all(requests);
    showToast();
} catch(err) {
    errorToast();
}
```